### ISSUE TITLE
Umístění bodových značek na linii v QGIS-u

### ISSUE THREAD

**Issue by raugustyn** [raugustyn/BetaNM#11]
@rmakovec: Druhý hlavní problém je s SVG značkou na linii, zatím jsem zkoušel dřevený plot a drátěný plot a značka se špatně nandá na linii

**Comment by rmakovec** [comment]
i když použiji správně umístění Hstřed a Vdole, tak jsou značky od linie posunuty

**Comment by rmakovec** [comment]
i ve vašem tutoriálu na obarvení značek je vidět,že se SVG značka 2_ plot dřevěný umístí špatně na linii i když máte zvoleno Hcenter a Bottom

**Comment by raugustyn** [comment]
No, uznávám že úplnou selskou logiku to nemá, ale nastavit to jde, viz projekt Heap/TestProject07.qgs.

![Nastvení plotu](https://raw.githubusercontent.com/vugtk21/BetaNM/7ae566ff13520d18b71b28ffda3dc478af3a7681/Heap/PlotNaLinii02.png)

![Značka nalepená na čáru](https://raw.githubusercontent.com/vugtk21/BetaNM/7ae566ff13520d18b71b28ffda3dc478af3a7681/Heap/PlotNaLinii01.png)

**Comment by rmakovec** [comment]
no logika tam není, jako dá se to nacvakat pro každou linii, ale tu bude dobrá opičárna pro tich 620 linii 2 krát

**Comment by rmakovec** [comment]
Tak jelikož jsem s Marianou neuspěl, tak dávám sem jako podklad pro řešení.
Na obrázku je to nejvíce vidět u linií s SVG značkou střídavě po obou stranách. Pro SVG značku na spodní straně linie je definováno odsazení na linii = poslední SVG značka na linii je za koncovým bodem linie.

Je nutné mít odsazení i pro první značku SVG na linií z důvodu, aby nezasahovala do bodové značky na lomovém bodu linie. Takže se v určitých případech (délky linie) zobrazí značka za lomovým bodem.

Pokud se to v QGISU neudělá, tak se musí vytvořit funkce na umísťování SVG značek, která vezme délku linie, odečte krajní odsazení (aby značky nebyli přímo v lomovém bodu) pro dané měřítko a dopočítá umístění značek. Funkcionalita jak pro QGIS tak pro mobilní aplikaci.

![image](https://cloud.githubusercontent.com/assets/6359293/7290914/c4c2b8fe-e987-11e4-831c-b27178e618de.png)

**Comment by raugustyn** [comment]
hm, vidím. nezapomeňte měnit kolonku asignee, aby bylo vidět že je míč na mojí straně.

jak jste vyrobit odkaz na https://cloud.githubusercontent.com/assets/6359293/7290914/c4c2b8fe-e987-11e4-831c-b27178e618de.png. To je něco uvnitř repozitáře nebo to ukazuje pryč?

**Comment by rmakovec** [comment]
prostě jsem přetáhl obrázek jako sem do textu

**Comment by raugustyn** [comment]
tvl

**Comment by rmakovec** [comment]
co?

**Comment by raugustyn** [comment]
Chybnou kresbu na obrázku výše vidím, ale nepodařilo se mi to nasimulovat. Asi by bylo vhodné sestavit adresář ve kterém by bylo jenom těch pár čar v shapefile a použitých značek, včetně projektu qgis, aby se s tím dalo pracovat. Jako že izolovat problém do separátního Issue.

**Comment by raugustyn** [comment]
Testovací data jsem udělal do https://github.com/vugtk21/BetaNM/raw/83fde4dab0c96f31eeb6af7c70560f6c099529fc/Heap/TestData.zip
